src/provider/controller/ToastManager.ts

```diff
@@ -1,78 +1,180 @@
-import {StoredNotification} from '../model/StoredNotification';
-
-import {WindowInfo} from './WindowInfo';
+import {PointTopLeft} from 'openfin/_v2/api/system/point';
+import {Rect} from 'openfin/_v2/api/system/monitor';
 
-interface Toast {
-    note: fin.OpenFinNotification;
-    meta: StoredNotification
-}
+import {StoredNotification} from '../model/StoredNotification';
+import {getNotificationCenterVisibility, getToastDirection} from '../store/ui/selectors';
+import {Toast, ToastEvent} from '../model/Toast';
+import {watchForChange} from '../store/utils/watch';
+import {Store} from '../store';
 
+export type WindowDimensions = {height: number, width: number};
 
-/**
- * @class ToastManager Handles all toasts
- */
 export class ToastManager {
-    private static singleton: ToastManager;
+    private static _instance: ToastManager;
+    private _store!: Store;
+    private _toasts: Map<string, Toast> = new Map();
+    private _stack: Toast[] = [];
+    private _bounds!: Required<Rect>;
 
-    private toasts: Toast[] = [];
+    private constructor() {
+    }
 
-    constructor() {
-        if (ToastManager.singleton) {
-            return ToastManager.singleton;
+    public static get instance(): ToastManager {
+        if (!ToastManager._instance) {
+            ToastManager._instance = new ToastManager();
         }
+        return ToastManager._instance;
+    }
+
+    public async initialize(store: Store): Promise<void> {
```

Comment:
This initialize pattern seems a bit weird.  I think I understand why, but wondering if we could shove it in the constructor... some how

Reply:
Also function docs

Reply:
Moved initialize into the constructor and removed AsyncInit.